### PR TITLE
Changes for automated emails

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
Issue #: https://github.com/bcgov/met-public/issues/

*Description of changes:*
- Added new email template and logic to trigger new email for engagements closing within next two days
- Changed email content for subscription confirmation and new engagement publish
- Added logic to check the users subscription and send appropriate email for new engagement publish and engagements closing within next two days

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/services/email_verification_service.py [hunk 1]
```diff
@@ -106,7 +108,8 @@ def _send_verification_email(email_verification: dict) -> None:
             raise ValueError('Engagement not found')
 
         subject, body, args, template_id = EmailVerificationService._render_email_template(
-            survey, email_verification.get('verification_token'), email_verification.get('type'), participant_id)
+            survey, email_verification.get('verification_token'),
```
Comment (jadmsaadaot) [inline met-api/src/met_api/services/email_verification_service.py:111]: may we please prettify the style here, one param in each line maybe

File: met-api/src/met_api/services/email_verification_service.py [hunk 2]
```diff
@@ -119,49 +122,50 @@ def _send_verification_email(email_verification: dict) -> None:
                 status_code=HTTPStatus.INTERNAL_SERVER_ERROR) from exc
 
     @staticmethod
-    def _render_email_template(survey: SurveyModel, token, email_type: EmailVerificationType, participant_id):
+    def _render_email_template(survey: SurveyModel, token, email_type: EmailVerificationType,
```
Comment (jadmsaadaot) [inline met-api/src/met_api/services/email_verification_service.py:125]: style

File: met-api/src/met_api/services/email_verification_service.py (hunk 1 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/services/email_verification_service.py]: updated the style, thanks.

File: met-api/src/met_api/services/email_verification_service.py (hunk 2 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/services/email_verification_service.py]: updated

**Review by djnunez-aot — APPROVED** [review]

**Pull request merged**